file-system/spec/spec.go

```diff
@@ -0,0 +1,19 @@
+// Package filesystemspec describes the interface of file system
+// implementations.
+package filesystemspec
+
+import (
+	"os"
+)
+
+// FileSystem implements file system operations.
+type FileSystem interface {
+	// ReadDir implements os.ReadDir.
```

Comment:
this godoc is misleading, as the references stdlib things are functions, not interfaces. How about "ReadDir is the equivalent to os.ReadDir"